### PR TITLE
Subscription cr apply

### PR DESCRIPTION
**Pull request by tharindu1st**
Subscription cr apply
#1951 https://github.com/wso2/apk/issues/1969

### CODE REVIEW

File: common-controller/internal/config/default_config.go [hunk 1]
```diff
@@ -35,5 +35,11 @@ var defaultConfig = &Config{
 		},
 		Environment:       "Default",
 		InternalAPIServer: internalAPIServer{Port: 18003},
+		ControlPlane: controlplane{
+			Enabled:       false,
+			Host:          "localhost",
+			EventPort:     18000,
+			RetryInterval: 5,
+			Persistence:   persistence{Type: "K8s"}},
```
Comment (AmaliMatharaarachchi) [inline common-controller/internal/config/default_config.go:43]: what about type ETCD?

File: common-controller/internal/controlplane/k8s_artifact_deployer.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+	return nil
+}
+
+// DeleteApplicationMappings deletes an application mapping
+func (k8sArtifactDeployer K8sArtifactDeployer) DeleteApplicationMappings(applicationMapping string) error {
+	crApplicationMapping := cpv1alpha2.ApplicationMapping{}
+	err := k8sArtifactDeployer.client.Get(context.Background(), client.ObjectKey{Name: applicationMapping, Namespace: utils.GetOperatorPodNamespace()}, &crApplicationMapping)
+	if err != nil {
+		if !k8error.IsNotFound(err) {
+			loggers.LoggerAPKOperator.ErrorC(logging.PrintError(logging.Error1102, logging.BLOCKER, "Failed to get application mapping from k8s %v", err.Error()))
```
Comment (AmaliMatharaarachchi) [inline common-controller/internal/controlplane/k8s_artifact_deployer.go:185]: I think we don't want to restart the adapter if we encounter such errors. logging.BLOCKER will crash the adapter. Use it carefully

Comment (AmaliMatharaarachchi) [reply common-controller/internal/controlplane/k8s_artifact_deployer.go:185]: Please fix in all places.

File: helm-charts/values.yaml [hunk 3]
```diff
@@ -143,8 +143,8 @@ wso2:
             failureThreshold: 5
           strategy: RollingUpdate
           replicas: 1
-          imagePullPolicy: Always
-          image: wso2/apk-common-controller:1.1.0-m2
+          imagePullPolicy: IfNotPresent
```
Comment (AmaliMatharaarachchi) [inline helm-charts/values.yaml:146]: I guess this is an unintentional change.

**Review by AmaliMatharaarachchi — CHANGES_REQUESTED** [review]

File: common-controller/internal/config/default_config.go (hunk 1 above)
Comment (tharindu1st) [reply common-controller/internal/config/default_config.go:43]: ideal name is K8s right since ETCD is implementation detail

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Review by AmaliMatharaarachchi — APPROVED** [review]

**Pull request merged**